src/TSTransformer/nodes/expressions/transformTemplateExpression.ts

```diff
@@ -1,29 +1,26 @@
 import luau from "@roblox-ts/luau-ast";
-import { errors } from "Shared/diagnostics";
-import { assert } from "Shared/util/assert";
 import { TransformState } from "TSTransformer";
-import { DiagnosticService } from "TSTransformer/classes/DiagnosticService";
 import { createStringFromLiteral } from "TSTransformer/util/createStringFromLiteral";
 import { ensureTransformOrder } from "TSTransformer/util/ensureTransformOrder";
-import { binaryExpressionChain } from "TSTransformer/util/expressionChain";
-import {
-	isDefinitelyType,
-	isLuaTupleType,
-	isPossiblyType,
-	isStringType,
-	isUndefinedType,
-} from "TSTransformer/util/types";
 import ts from "typescript";
 
-export function transformTemplateExpression(state: TransformState, node: ts.TemplateExpression) {
-	// if there are zero templateSpans, this must be a ts.NoSubstitutionTemplateLiteral
-	// and will be handled in transformStringLiteral
-	assert(node.templateSpans.length > 0);
+function createInterpolatedStringPart(node: ts.TemplateLiteralToken | ts.StringLiteral) {
+	return luau.create(luau.SyntaxKind.InterpolatedStringPart, { text: createStringFromLiteral(node) });
+}
+
+// backtick string literals without interpolation expressions should be preserved
+// as they still are valid in luau
+export function transformNoSubstitutionTemplateLiteral(state: TransformState, node: ts.NoSubstitutionTemplateLiteral) {
+	return luau.create(luau.SyntaxKind.InterpolatedString, {
+		parts: luau.list.make(createInterpolatedStringPart(node)),
+	});
+}
```

Comment:
I think this should live in it's own file. We rarely have multiple transformX functions in the same file.